app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -106,6 +106,7 @@ public void onReceive(Context context, Intent intent) {
     private final BroadcastReceiver showUpdateDialog = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
+            assert BuildConfig.isInternetAvailable;
```

Comment:
I wanted to ensure the *failing* stubs I added aren't called in an offline build—note they throw exceptions.

That's only true because these assertions hold. And checking that's true took some non-local reasoning and intent chasing, so I thought I'd better document it.

But indeed `MainActivity` could skip the check if `AppUpdateUtil.checkForUpdate` was a no-op in the offline version.

And it sounds like we shouldn't use this flag in `app/src/online/java/com/kamron/pogoiv/CrashlyticsWrapper.java` which is already online-only 😉 . 

I suspect I'd keep using the flag in `app/src/main/java/com/kamron/pogoiv/SettingsActivity.java` though.